src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -69,7 +79,28 @@ class TaskKiller @Inject() (
     }
   }
 
-  private[this] def expunge(instances: Seq[Instance]): Future[Done] = {
+  private[this] def awaitConsideredTerminal(instanceIds: Set[Instance.Id]): Source[Set[Instance.Id], NotUsed] =
+    instanceTracker.instanceUpdates
+      .flatMapConcat {
+        case (snapshot, updates) =>
+          val pendingInstances: Set[Instance.Id] = snapshot.instances.collect {
```

Comment:
Hm, this is tricky. What about terminal instances with `Goal.Running`?

Reply:
That's a great catch; those instances will be restarted and we will need to wait for the goal and the status to be terminal